File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/watch/execution_watch_request.py
from __future__ import annotations

from dataclasses import dataclass

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions
from .execution_watch_channel import AiSdkExecutionWatchChannel


@dataclass(frozen=True)
class AiSdkExecutionWatchRequest(AiSdkWireModel):
    execution_id: str
    schema_version: int = AiSdkSchemaVersions.EXECUTION_WATCH_REQUEST
    channels: tuple[AiSdkExecutionWatchChannel, ...] = ()
    after_sequence: int | None = None
    include_initial_snapshot: bool = True

File: implementations/node/sdk/src/contracts/watch/execution-watch-request.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";
import type { AiSdkExecutionWatchChannel } from "./execution-watch-channel.js";

/** Portable watch request. An empty channels array means all public channels. */
export interface AiSdkExecutionWatchRequest {
  readonly schemaVersion?: typeof AI_SDK_SCHEMA_VERSIONS.executionWatchRequest;
  readonly executionId: string;
  readonly channels?: readonly AiSdkExecutionWatchChannel[];
  readonly afterSequence?: number;
  readonly includeInitialSnapshot?: boolean;
}
